apps/staking/app/layout.tsx

```diff
@@ -25,6 +29,12 @@ export default async function RootLayout({ children }: { children: ReactNode })
   const { locale, direction, messages } = await getLocalizationData();
   const wagmiCookie = (await headers()).get('cookie');
 
+  /**
+   *  We don't need to handle any errors from the remote flag functions as any errors are handled
+   *  in the function call and call happens on the same thread as this is server-side rendered.
+   */
+  const enabledFlags = new Set((await getRemoteFeatureFlags()).flags);
```

Comment:
I had a look at `getRemoteFeatureFlagsGeneric` and I think I am missing some better sqlite context. What makes that function async? openDatabase doesn't look to be async either?

Reply:
has to be async as its a server action, we want to call it from the client sometimes

Reply:
Cool resolving